<launch>
    <arg name="use_sim_time" default="false"/>

    <arg name="rviz" default="true"/>
    <arg name="map" default=""/>

    <arg name="oneshot" default="false"/> <!-- disregard path updates unless navigation is restarted -->

    <arg name="telemetry" default="true"/> <!-- important telemetry logging to separate topic -->

    <arg name="df_threshold" default="1.0" />
    <node name="path_frechet" pkg="central_nav" exec="path_frechet_node" output="screen">
        <remap from="poses" to="/robot_poses" />
        <remap from="paths_in" to="/robot_paths/raw" /> <!-- not really raw though, but it's as raw as we can get, as far as the central system is concerned -->
        <remap from="paths_out" to="/robot_paths" />
        <param name="threshold" value="$(var df_threshold)" />
        <param name="use_sim_time" value="$(var use_sim_time)" />
    </node>

    <arg name="min_path_dist" default="0.1" />
    <arg name="ix_radius" default="0.6" />
    <node name="ix_detect" pkg="central_nav" exec="ix_detect_node" output="screen">
        <remap from="paths" to="/robot_paths" />
        <remap from="intersections" to="/ix_markers/detected" />
        <param name="min_path_dist" value="$(var min_path_dist)" />
        <param name="ix_radius" value="$(var ix_radius)" />
        <param name="use_sim_time" value="$(var use_sim_time)" />
    </node>

    <node name="central_node" pkg="central_nav" exec="central_node" output="screen">
        <remap from="robot_poses" to="/robot_poses"/>
        <remap from="ix_markers_in" to="/ix_markers/detected"/>
        <remap from="ix_markers_out" to="/ix_markers"/>
        <remap from="robot_pass" to="/robot_pass"/>
        <remap from="robot_stop" to="/robot_stop"/>
        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <param name="oneshot" value="$(var oneshot)"/>    

        <param name="telemetry" value="$(var telemetry)"/>
        <remap from="telemetry" to="/telemetry"/>
    </node>

    <!-- <include file="$(find-pkg-share central_nav)/launch/telemetry_launch.xml" if="$(var telemetry)">
        <arg name="use_sim_time" value="$(var use_sim_time)" />
    </include> -->

    <arg name="visualise" default="true" />
    <include file="$(find-pkg-share central_nav)/launch/visualisation_launch.xml" if="$(var visualise)">
        <arg name="rviz" value="$(var rviz)" />
        <arg name="map" value="$(var map)" />
        <arg name="use_sim_time" value="$(var use_sim_time)" />
    </include>
</launch>